autoscheduler/frontend/src/redux/actions/courseCards.ts

```diff
@@ -212,3 +234,71 @@ export function updateCourseCard(index: number, courseCard: CourseCardOptions, t
     return dispatch(updateCourseCardSync(index, courseCard));
   };
 }
+
+export function clearCourseCards(): ClearCourseCardsAction {
+  return { type: CLEAR_COURSE_CARDS };
+}
+
+/**
+ * Helper function to fetch updated sections for a course card and keep selected sections
+ * @param courseCard course card to update
+ * @param term term to fetch sections for
+ */
+async function updateSectionsForCourseCard(
+  courseCard: CourseCardOptions,
+  term: string,
+): Promise<CourseCardOptions> {
+  return createCourseCardFrom(courseCard, term).then((newCourseCard) => {
+    // course now updated with new sections, re-select sections from original courseCard
+    const selectedSections = new Set(courseCard.sections?.filter(({ selected }): boolean => (
+      selected
+    )).map((section): number => section.section.id));
+
+    newCourseCard.sections.forEach((section) => {
+      // eslint-disable-next-line no-param-reassign
+      if (selectedSections.has(section.section.id)) section.selected = true;
+    });
+
+    return newCourseCard;
+  });
+}
+
+/**
+ * Creates an updated course card from courseCard by making an API call for updated
+ * section information
+ * @param courseCard Course card to update
+ * @param term Term to fetch section information from
+ */
+export function replaceCourseCards(
+  courseCards: CourseCardArray,
+  term: string,
+): ThunkAction<void, RootState, undefined, CourseCardAction> {
+  // data for sections might have changed since last visit, so create a new course card
+  // for each one asynchronously and update them with data from courseCards
+  return (dispatch): void => {
+    // determine which cards actually exist
+    const cards = [];
+    for (let i = 0; i < courseCards.numCardsCreated; i++) {
+      const card = courseCards[i];
+      if (card) cards.push(card);
+    }
+
+    // create promise for each updated course card
+    const newCourseCards = cards.map((card) => (updateSectionsForCourseCard(card, term)));
+
+    // clear all course cards
+    new Promise((resolve) => { dispatch(clearCourseCards()); resolve(); }).then(async () => {
+      // add empty course card if there are no course cards for the current term
+      if (cards.length === 0) {
+        dispatch(addCourseCard());
+      } else {
+        // update information from all course cards and add them in order
+        for (let i = 0; i < newCourseCards.length; i++) {
+          // eslint-disable-next-line no-await-in-loop
+          const newCourseCard = await newCourseCards[i];
```

Comment:
Why do we need an `await` here? Array access isn't asynchronous, is it?

Reply:
We want the courses to be in the same order as they're stored in when they're recreated, so here it has to wait for the previous card to be generated before dispatching to add the next card